integrations/azure_ai_search/src/haystack_integrations/document_stores/azure_ai_search/document_store.py

```diff
@@ -0,0 +1,450 @@
+        :param kwargs: Optional keyword parameters.
+        """
+
+        # default fields to create index based on Haystack Document (id, content, embedding)
+        default_fields = [
+            SimpleField(name="id", type=SearchFieldDataType.String, key=True, filterable=True),
+            SearchableField(name="content", type=SearchFieldDataType.String),
+            SearchField(
+                name="embedding",
+                type=SearchFieldDataType.Collection(SearchFieldDataType.Single),
+                searchable=True,
+                hidden=False,
+                vector_search_dimensions=self._embedding_dimension,
+                vector_search_profile_name="default-vector-config",
+            ),
+        ]
+
+        if not index_name:
+            index_name = self._index_name
+        if self._metadata_fields:
+            default_fields.extend(self._create_metadata_index_fields(self._metadata_fields))
+        index = SearchIndex(
+            name=index_name, fields=default_fields, vector_search=self._vector_search_configuration, **kwargs
+        )
+        if self._index_client:
+            self._index_client.create_index(index)
+
+    def to_dict(self) -> Dict[str, Any]:
+        # This is not the best solution to serialise this class but is the fastest to implement.
+        # Not all kwargs types can be serialised to text so this can fail. We must serialise each
+        # type explicitly to handle this properly.
+        """
+        Serializes the component to a dictionary.
+
+        :returns:
+            Dictionary with serialized data.
+        """
+        return default_to_dict(
+            self,
+            azure_endpoint=self._azure_endpoint.to_dict() if self._azure_endpoint is not None else None,
+            api_key=self._api_key.to_dict() if self._api_key is not None else None,
+            index_name=self._index_name,
+            create_index=self._create_index,
+            embedding_dimension=self._embedding_dimension,
+            metadata_fields=self._metadata_fields,
+            vector_search_configuration=self._vector_search_configuration.as_dict(),
+            **self._kwargs,
+        )
+
+    @classmethod
+    def from_dict(cls, data: Dict[str, Any]) -> "AzureAISearchDocumentStore":
+        """
+        Deserializes the component from a dictionary.
+
+        :param data:
+            Dictionary to deserialize from.
+
+        :returns:
+            Deserialized component.
+        """
+
+        deserialize_secrets_inplace(data["init_parameters"], keys=["api_key", "azure_endpoint"])
+        if (vector_search_configuration := data["init_parameters"].get("vector_search_configuration")) is not None:
+            data["init_parameters"]["vector_search_configuration"] = VectorSearch.from_dict(vector_search_configuration)
+        return default_from_dict(cls, data)
+
+    def count_documents(self, **kwargs: Any) -> int:
```

Comment:
The DocumentStore protocol is not respected due to the presence of kwargs
See https://github.com/deepset-ai/haystack/blob/13804293b1bb79743e5a30e980b76a0561dcfaf8/haystack/document_stores/types/protocol.py#L37